### ISSUE TITLE
error on building

### ISSUE THREAD

**Issue by theofficialgman** [input-leap/input-leap#1673]
### What happened?

attempting to build following the build workflow I run into an error with building input-leap

build commands:
```bash
cd /tmp
sudo apt-get install -y \
                  cmake \
                  g++ \
                  git \
                  libavahi-compat-libdnssd-dev \
                  libcurl4-openssl-dev \
                  libice-dev \
                  libsm-dev \
                  libssl-dev \
                  libxinerama-dev \
                  libxrandr-dev \
                  libxtst-dev \
                  libxkbcommon-dev \
                  libglib2.0-dev \
                  ninja-build \
                  qtdeclarative5-dev \
                  qttools5-dev \
                  meson

git clone --recursive https://github.com/input-leap/input-leap.git
cd input-leap/

sudo apt-get install -y libsystemd-dev meson git ca-certificates python3-pytest python3-attr python3-dbusmock python3-structlog python3-jinja2
git clone --depth 1 --branch 0.99.1 https://gitlab.freedesktop.org/libinput/libei
cd libei
meson -Dprefix=/usr -Dtests=disabled -Dliboeffis=disabled -Ddocumentation=[] _libei_builddir
ninja -C _libei_builddir
sudo ninja -C _libei_builddir install
cd ..
rm -rf libei

sudo apt-get install -y libglib2.0 gettext dbus meson libgirepository1.0-dev libgtk-3-dev valac python3-pip python3-dbusmock
git clone https://github.com/whot/libportal
cd libportal
git checkout wip/inputcapture
meson -Dprefix=/usr -Dbackend-gtk3=enabled -Ddocs=false _libportal_builddir
ninja -C _libportal_builddir
sudo ninja -C _libportal_builddir install
cd ..
rm -rf libportal

B_CMAKE_FLAGS="-DINPUTLEAP_BUILD_LIBEI=TRUE" ./clean_build.sh
```

### Version

From Git HEAD or commit (specify below)

### Git commit hash (if applicable)

6446288aa619133f287cf3166bc924a99d073ea2

### What OSes are you seeing the problem on? (Check all that apply)

Linux

### What OS versions are you using?

Debian Bookworm

### Relevant log output

```shell
/usr/bin/ninja
Starting Input Leap Debug build in 'build'...
-- The C compiler identification is GNU 12.2.0
-- The CXX compiler identification is GNU 12.2.0
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /usr/bin/cc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /usr/bin/c++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Full InputLeap version string is '2.4.0-release-release'
-- Found PkgConfig: /usr/bin/pkg-config (found version "1.8.1") 
-- Looking for include file sys/socket.h
-- Looking for include file sys/socket.h - found
-- Looking for include file sys/utsname.h
-- Looking for include file sys/utsname.h - found
-- Looking for getpwuid_r
-- Looking for getpwuid_r - found
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Success
-- Found Threads: TRUE  
-- Found CURL: /usr/lib/aarch64-linux-gnu/libcurl.so (found version "7.88.1")  
-- Checking for module 'avahi-compat-libdns_sd'
--   Found avahi-compat-libdns_sd, version 0.8
-- Checking for modules 'x11;xext;xrandr;xinerama;xtst;xi'
--   Found x11, version 1.8.4
--   Found xext, version 1.3.4
--   Found xrandr, version 1.5.2
--   Found xinerama, version 1.1.4
--   Found xtst, version 1.2.3
--   Found xi, version 1.8
-- Checking for modules 'ice;sm'
--   Found ice, version 1.0.10
--   Found sm, version 1.2.3
-- Checking for module 'libei-1.0 >= 0.99.1'
--   Found libei-1.0 , version 0.99.1
-- Checking for module 'xkbcommon'
--   Found xkbcommon, version 1.5.0
-- Checking for modules 'glib-2.0;gio-2.0'
--   Found glib-2.0, version 2.74.6
--   Found gio-2.0, version 2.74.6
-- Checking for module 'libportal'
--   Found libportal, version 0.6
-- Looking for xdp_session_connect_to_eis
-- Looking for xdp_session_connect_to_eis - found
-- Looking for xdp_input_capture_session_connect_to_eis
-- Looking for xdp_input_capture_session_connect_to_eis - found
-- Performing Test HAVE_LIBPORTAL_OUTPUT_NONE
-- Performing Test HAVE_LIBPORTAL_OUTPUT_NONE - Success
-- Looking for include file dns_sd.h
-- Looking for include file dns_sd.h - found
-- Configuring directory /tmp/input-leap/build/rpm
-- Configuring file input-leap.spec
-- Configuring done
-- Generating done
-- Build files have been written to: /tmp/input-leap/build
[82/256] Building CXX object src/lib/platform/CMakeFiles/platform.dir/PortalRemoteDesktop.cpp.o
FAILED: src/lib/platform/CMakeFiles/platform.dir/PortalRemoteDesktop.cpp.o 
/usr/bin/c++ -DINPUTLEAP_BUILD_DATE=\"20230729\" -DINPUTLEAP_BUILD_NUMBER=1 -DINPUTLEAP_REVISION=\"6446288a\" -DINPUTLEAP_VERSION=\"2.4.0-release\" -DINPUTLEAP_VERSION_MAJOR=2 -DINPUTLEAP_VERSION_MINOR=4 -DINPUTLEAP_VERSION_PATCH=0 -DINPUTLEAP_VERSION_STRING=\"2.4.0-release-release\" -DSYSAPI_UNIX=1 -DWINAPI_LIBEI=1 -DWINAPI_XWINDOWS=1 -I/usr/include/libei-1.0 -I/usr/include/glib-2.0 -I/usr/lib/aarch64-linux-gnu/glib-2.0/include -I/usr/include/libmount -I/usr/include/blkid -I/usr/include/gio-unix-2.0 -I/tmp/input-leap/src/./lib -I/tmp/input-leap/build/src/lib -isystem /usr/include/avahi-compat-libdns_sd -isystem /tmp/input-leap/ext/gulrak-filesystem/include -g -fPIC -std=c++14 -MD -MT src/lib/platform/CMakeFiles/platform.dir/PortalRemoteDesktop.cpp.o -MF src/lib/platform/CMakeFiles/platform.dir/PortalRemoteDesktop.cpp.o.d -o src/lib/platform/CMakeFiles/platform.dir/PortalRemoteDesktop.cpp.o -c /tmp/input-leap/src/lib/platform/PortalRemoteDesktop.cpp
/tmp/input-leap/src/lib/platform/PortalRemoteDesktop.cpp: In member function ‘void inputleap::PortalRemoteDesktop::cb_session_started(GObject*, GAsyncResult*)’:
/tmp/input-leap/src/lib/platform/PortalRemoteDesktop.cpp:122:36: error: too few arguments to function ‘int xdp_session_connect_to_eis(XdpSession*, GError**)’
  122 |     fd = xdp_session_connect_to_eis(session);
      |          ~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~
In file included from /usr/include/libportal/portal.h:36,
                 from /tmp/input-leap/src/./lib/platform/PortalRemoteDesktop.h:26,
                 from /tmp/input-leap/src/lib/platform/PortalRemoteDesktop.cpp:18:
/usr/include/libportal/remote.h:181:11: note: declared here
  181 | int       xdp_session_connect_to_eis    (XdpSession  *session,
      |           ^~~~~~~~~~~~~~~~~~~~~~~~~~
[83/256] Building CXX object src/lib/platform/CMakeFiles/platform.dir/PortalInputCapture.cpp.o
FAILED: src/lib/platform/CMakeFiles/platform.dir/PortalInputCapture.cpp.o 
/usr/bin/c++ -DINPUTLEAP_BUILD_DATE=\"20230729\" -DINPUTLEAP_BUILD_NUMBER=1 -DINPUTLEAP_REVISION=\"6446288a\" -DINPUTLEAP_VERSION=\"2.4.0-release\" -DINPUTLEAP_VERSION_MAJOR=2 -DINPUTLEAP_VERSION_MINOR=4 -DINPUTLEAP_VERSION_PATCH=0 -DINPUTLEAP_VERSION_STRING=\"2.4.0-release-release\" -DSYSAPI_UNIX=1 -DWINAPI_LIBEI=1 -DWINAPI_XWINDOWS=1 -I/usr/include/libei-1.0 -I/usr/include/glib-2.0 -I/usr/lib/aarch64-linux-gnu/glib-2.0/include -I/usr/include/libmount -I/usr/include/blkid -I/usr/include/gio-unix-2.0 -I/tmp/input-leap/src/./lib -I/tmp/input-leap/build/src/lib -isystem /usr/include/avahi-compat-libdns_sd -isystem /tmp/input-leap/ext/gulrak-filesystem/include -g -fPIC -std=c++14 -MD -MT src/lib/platform/CMakeFiles/platform.dir/PortalInputCapture.cpp.o -MF src/lib/platform/CMakeFiles/platform.dir/PortalInputCapture.cpp.o.d -o src/lib/platform/CMakeFiles/platform.dir/PortalInputCapture.cpp.o -c /tmp/input-leap/src/lib/platform/PortalInputCapture.cpp
/tmp/input-leap/src/lib/platform/PortalInputCapture.cpp: In member function ‘void inputleap::PortalInputCapture::cb_init_input_capture_session(GObject*, GAsyncResult*)’:
/tmp/input-leap/src/lib/platform/PortalInputCapture.cpp:143:55: error: too few arguments to function ‘int xdp_input_capture_session_connect_to_eis(XdpInputCaptureSession*, GError**)’
  143 |     auto fd = xdp_input_capture_session_connect_to_eis(session);
      |               ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~
In file included from /usr/include/libportal/portal.h:30,
                 from /tmp/input-leap/src/./lib/platform/PortalInputCapture.h:28,
                 from /tmp/input-leap/src/lib/platform/PortalInputCapture.cpp:17:
/usr/include/libportal/inputcapture.h:100:12: note: declared here
  100 | int        xdp_input_capture_session_connect_to_eis (XdpInputCaptureSession  *session,
      |            ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
[87/256] Building CXX object src/lib/server/CMakeFiles/server.dir/ClientListener.cpp.o
ninja: build stopped: subcommand failed.
```

**Comment by shymega** [comment]
We need more eyes on this. Looks like function args weren't updated following upstream.

Cc: @whot.

**Comment by whot** [comment]
fix for this is pending in #1670

**Comment by shymega** [comment]
Of course. I forgot about that. Bypassed the review, and now merged.

@theofficialgman Please try pulling latest mainline, and building again.

**Comment by theofficialgman** [comment]
@shymega yes that fixed building.
but how broken is input-leap expected to be on wayland right now?
on the server I get this spammed in log over and over
```
2023-07-30T12:18:47] INFO: config file: /tmp/InputLeap.qdhxDa
[2023-07-30T12:18:47] INFO: log level: INFO
[2023-07-30T12:18:47] ERROR: Failed to initialize InputCapture session, quitting: GDBus.Error:org.freedesktop.DBus.Error.UnknownMethod: No such interface “org.freedesktop.portal.InputCapture” on object at path /org/freedesktop/portal/desktop
	/tmp/input-leap/src/lib/platform/PortalInputCapture.cpp,135
..... 9: 0xff1b (Escape) level=0 modmask: 000000 (sensitive: 000000)
..... 10: 0x31 (1) level=0 modmask: 000000 (sensitive: 000000)
..... 10: 0x21 (exclam) level=1 modmask: 0x0001 (sensitive: 0x0001)
..... 11: 0x32 (2) level=0 modmask: 000000 (sensitive: 000000)
..... 11: 0x40 (at) level=1 modmask: 0x0001 (sensitive: 0x0001)
..... 12: 0x33 (3) level=0 modmask: 000000 (sensitive: 000000)
..... 12: 0x23 (numbersign) level=1 modmask: 0x0001 (sensitive: 0x0001)
started server (IPv4/IPv6), waiting for clients
 12:18:47 | DEBUG | queuing event type EI_EVENT_DISCONNECT (2)
      ... | [0;1;39mDEBUG[0m | deregistering ei_handshake v1 object 0
[2023-07-30T12:18:48] ERROR: process exited with error code: 11
[2023-07-30T12:18:48] INFO: detected process not running, auto restarting
```
adn clicking the "stop" button crashes my compositor (kwin-wayland) and I have to restart the system.

**Comment by shymega** [comment]
Yeah, Wayland isn't ready yet. I don't know why you're attempting to use it right now, but it's not advisable. Nor is building from Git - hence you get bugs like the ones you're experiencing.

**Comment by theofficialgman** [comment]
oh I was not aware that `xdg-desktop-portals` also needed support for this on the system even when using this on the host. I thought libei was enough.

guess I also need to build `https://github.com/whot/xdg-desktop-portal/tree/wip/capture-input`

**Comment by theofficialgman** [comment]
I've built and installed that and still the same issue.
Am I missing something? Do all the individual `xdg-desktop-portal` backends need to implement and adopt this before it will work?

**Comment by shymega** [comment]
No, you need to wait until it's merged upstream. Also, each backend needs to implement the interface. This is not something that's ready yet.

For now, you'll need to stick with X11.

**Comment by shymega** [comment]
I'm gonna go ahead and close this issue, because the underlying cause is outside of our control.

**Comment by theofficialgman** [comment]
> No, you need to wait until it's merged upstream. Also, each backend needs to implement the interface. This is not something that's ready yet.

So the answer is Yes, not No.

Killing the compositor on failure doesn't seem like a great result if the interface isn't present. Maybe that should be looked into.

**Comment by shymega** [comment]
I'm referring to the InputCapture interface. You need to wait until it's merged upstream and also wait for respective backends to support the InputCapture interface.

It's all a WIP, so this is expected behaviour IMO.

**Comment by theofficialgman** [comment]
Does the client also require the InputCapture interface, I get the same error there on a wayland client? that does not makes sense to me. I thought that libei was being used for

So do both the server and client need the InputCapture portal implemented for wayland on each side then?

**Comment by shymega** [comment]
I can't remember. I'm mostly working on the Rust side of things.

**Comment by whot** [comment]
> Does the client also require the InputCapture interface, I get the same error there on a wayland client?

the server needs the InputCapture portal, the client the RemoteDesktop portal. There's a chance the code is buggy and tries to initialize both? It is guarded by `is_primary` in `EiScreen.cpp` though.

**Comment by shymega** [comment]
Reopening, potential bug? Input Leap shouldn't crash if it tries to use an non-existent portal.

**Comment by whot** [comment]
so, I just tried this and my observed behaviour on a machine *without* support for either portal is that both server and client correctly connect to the respective portal and then exit if the portal is not supported. This seems like it's the correct behaviour?

The problem is this line: `[2023-07-30T12:18:48] INFO: detected process not running, auto restarting`, so something in inputleap keeps restarting the server process (code is in `MainWindow.cpp`). unlike X11, arguably this should not happen - if we fail to connect to either portal this means we don't have DBus available or don't have the right portals installed. In both cases this isn't really recoverable and restarting won't magically make the session bus appear or support the new features.

**Comment by shymega** [comment]
Hm... OK, so that's the watchdog in the UI doing that. 

I don't think we handle that very well anyway - it's what I'm trying to fix in my fork, but I know that's not ready yet. 

Having it fallback to X11 would be preferable, right?

**Comment by whot** [comment]
> Hm... OK, so that's the watchdog in the UI doing that.

somewhat unrelated - this could/should be a systemd service instead? maybe best to discuss in a separate issue.

> Having it fallback to X11 would be preferable, right?

atm you have to pass `--use-ei` for this to work anyway because x11 is the default. fallback to x11 isn't necessarily useful either because that doesn't work under xwayland (though technically it could, now with xwayland supporting XTEST :).

**Comment by shymega** [comment]
> > Hm... OK, so that's the watchdog in the UI doing that.
> 
> somewhat unrelated - this could/should be a systemd service instead? maybe best to discuss in a separate issue.

This has been discussed before, but my understanding is that the UI sets up 'pipes' to `input-leapc` and `input-leaps` to communicate. When those pipes fail, or get killed, the UI is meant to restart them.

I think there's been a discussion on using systemd to start Input Leap at display-manager time.

> > Having it fallback to X11 would be preferable, right?
> 
> atm you have to pass `--use-ei` for this to work anyway because x11 is the default. fallback to x11 isn't necessarily useful either because that doesn't work under xwayland (though technically it could, now with xwayland supporting XTEST :).

Hm, OK.

**Comment by whot** [comment]
> my understanding is that the UI sets up 'pipes' to input-leapc and input-leaps to communicate. When those pipes fail, or get killed, the UI is meant to restart them.

sort-of, a quick check of the code shows it appears to be using [`QProcess::finished`](https://doc.qt.io/qt-6/qprocess.html#finished) to get notified when the child process dies. Which is actually a much nicer solution than having to use pipes (and I suspect Qt internally implements this with some variation of `wait()`)

**Comment by shymega** [comment]
@whot It was my understanding the GUI uses pipes, tbh - we had a few issues years back regarding this. Also, isn't `QProcess` just an abstraction over stdin/stdout comms?

**Comment by whot** [comment]
tbh I can't really say without digging more into the code (and I'm lacking the time for that atm). at least `git grep` doesn't find a `pipe()` call in the gui code. The [`QProcess` docs](https://doc.qt.io/qt-6/qprocess.html#details) suggest that there are pipes/sockets to communicate with the subprocess (i.e. send to stdin or read from stdout/stderr) but the actual checking if it's killed would probably be implemented via `wait()` or its derivatives (which tbh is the *right* way to do this). But again, this is assuming only, I have not checked the Qt sources and only done a cursory guess of the input-leap gui sources.

**Comment by shymega** [comment]
> tbh I can't really say without digging more into the code (and I'm lacking the time for that atm). at least `git grep` doesn't find a `pipe()` call in the gui code. The [`QProcess` docs](https://doc.qt.io/qt-6/qprocess.html#details) suggest that there are pipes/sockets to communicate with the subprocess (i.e. send to stdin or read from stdout/stderr) but the actual checking if it's killed would probably be implemented via `wait()` or its derivatives (which tbh is the _right_ way to do this). But again, this is assuming only, I have not checked the Qt sources and only done a cursory guess of the input-leap gui sources.

Ah, okay. I will query this with Povilas. I thought it used pipes.

**Comment by prahal** [comment]
Tested with both server and client input-leap on Wayland gnome-shell 45-rc1.
The only issue left so far is my client has two monitors and the cursor from the server can move on the primary screen of the input-leap client but get "stuck" on the border of the input-leap client secondary monitor (well I can move the cursor vertically on the secondary screen border but not horizontally).

**Comment by shymega** [comment]
On Wed, Sep 13, 2023 at 01:03:11PM -0700, Alban Browaeys wrote:
> Tested with both server and client input-leap on Wayland gnome-shell 45-rc1.
> The only issue left so far is my client has two monitors and the cursor from the server can move on the primary screen of the input-leap client but get "stuck" on the border of the input-leap client secondary monitor (well I can move the cursor vertically on the secondary screen border but not horizontally).

@whot Do you think this is related to `libei` or perhaps an Input Leap-ism?

**Comment by shymega** [comment]
On Wed, Sep 13, 2023 at 07:33:32AM -0700, Alban Browaeys wrote:
> With Debian rc-buggy mutter and xdg-desktop-portal-gnome, Debian unstable libei1 and Debian unstable libportal1 rebuild with the three commits in https://github.com/flatpak/libportal/compare/main...whot:libportal:wip/inputcapture added, I have input-leap working on wayland as a server and working (as these are not even in upstream master libportal).
> I build input-leap master commit ad3c0e32672cc1916173ac5539bf80104d12c9f8
> with `cmake ../ -DINPUTLEAP_BUILD_LIBEI=1`
> then `make` and `sudo make install`, this in a build subdir.
> 
> I noticed that if I do not have the libei/inputcapture support built into input-leap (I have to double check which one or both) I end up having the display server crash (confirmed on gome-shell 45-rc1 wayland, and I recall I had Xorg crashes even without building the wayland support into input-leap at least sometimes).
> I believe input-leap crashing either gnome-shell wayland or Xorg/gnome-shell is not an input-leap issue (a client program should not be able to crash the server even feeding bad inputs or calling the API incorrectly).
> Note that once the wayland support is compiled in and the libportal inputcapture dependencies are installed no more display server crashes. So the crashes might well be in the portal code.
> 
> I still have to check the wayland support in input-leap works on the client side.
> 
> I have an issue that is on the input-leap server input-leap asks for portal permission each time it starts. Could the setting be saved?
> 
> ```
> apt install -t rc-buggy mutter  gnome-shell libglib2.0-0
> sudo apt install -t unstable  libportal1 libportal-gtk4-1 libportal-gtk3-1 libportal-dev libportal-gtk4-dev xdg-desktop-portal xdg-desktop-portal-gnome
> ```
> 
> (mind with gnome 45 most extensions will need rework and thus stop working as of now)
> 
> rebuild libportal1 with dpkg-buildpackage after quilt importing the three wip commits told above.

This is worth reporting on the `libei` GitLab, if you're certain it's not Input Leap.

Bizarre issue, though.

Best wishes,
--
Dom Rodriguez
GPG Fingerprint: EB0D 45E6 D0DC 1BA1 A2B5  FC24 72DC F123 1E54 BD43

**Comment by prahal** [comment]
> This is worth reporting on the `libei` GitLab, if you're certain it's not Input Leap.

Might take time. Increasing the debug level of input-leap logs showed me the mouse positions.
Since then I changed my multimonitor setup in that I set the external monitor as primary instead of the laptop built-in one and inverted their relative position in gnome settings.
My latest tests since then had a different though still buggy behavior. input-leap client log properly shows the position of the mouse on all client's monitors. But the now secondary built-in display now properly enters the keyboard inputs to the text editors on this secondary built-in display but the fakeMouse events have no effect (even though they are logged in the input-leap client's log). Note that as in previous reports, I made initially these fakeMouse events  (like right click) are working on this secondary display. Note that the cursor is also not drawn on the secondary display (but is on the primary display of the input-display client).
Could the mouse cursor detection be libei's fault?

I still have to double-check the logs to be confident the logs are the same on the client when the cursor is shown and fake mouse events are working on the primary display of the client and when they are not (the debug2 logs are chatty).

**Comment by shymega** [comment]
I think it's hard to tell at this stage. What I would suggest is, replicating the same configuration, but on X11.

If that works, we can *partially* rule out Input Leap. We can rule out it being related to X11, but something to do with Wayland instead. If that's the case, we should then proceed to a more in-depth debugging, and work with @whot to identify the bug, which could, possibly, be outside of Input Leap's remit.

**Comment by prahal** [comment]
@shymega it is working fine with the input-leap server (mono monitor) as Wayland gnome-shell 45-rc1 and input-leap client Xorg (double monitor) Xorg gnome-shell 45-rc1 (I test with current input-leap and gnome-shell, though I had it working before I tried the client as Wayland too as I told in an above comment that input-leap wayland basics were functional).
I also had the same setup with both Xorg working before with the same monitor layout (at least with gnome 44).

**Comment by shymega** [comment]
@prahal Thanks, this is valuable info.

**Comment by prahal** [comment]
@shymega seems I cannot reproduce the multimonitor bug
Since when I had this bug I upgraded libei1 and libeis1 from 1.0.901-2 to  1.1.0-1 on Debian.
Will downgrade when time permits to confirm.

Is the non-working clipboard sharing in wayland mode expected?

**Comment by shymega** [comment]
> @shymega seems I cannot reproduce the multimonitor bug Since when I had this bug I upgraded libei1 and libeis1 from 1.0.901-2 to 1.1.0-1 on Debian. Will downgrade ei related packages when time permits to confirm. I did not updated input-leap code since then. I also updated gnome-shell and mutter from rc1 to 45.0 releases.

This is interesting. It could be the upgrade for `libei` then.

> Is the non-working clipboard sharing in wayland mode expected?

Clipboard is a mess. It doesn't really work on X11 either. My position on it is a refactor.

**Comment by prahal** [comment]
@shymega tested the reproducibility of the multimonitor bug on the input-leap client.
TLDR; mutter 45 rc-2 on the multi-monitors input leap client is the cause of the bug.

Downgrading mutter (libmutter-13-0 and its gir and common-bin companions) from 45.0 to 45-rc-2 on the Debian input-leap client reproduces the issue.
All the other packages make no difference: gnome-shell, libei1, libeis1. Be it on the input-leap server or client. I can also have mutter 45-rc-2 on the input-leap server and mutter 45.0 on the input-leap client and then the bug is still not reproducible.

I also finally tried gnome-shell 45-rc1/mutter 45-rc-2 on both the client and server (as in my initial report) but this time with gnome-shell Xorg on the input-leap server and gnome-shell wayland on the input-leap server. The bug is still there (having mutter 45-rc-2 on the client is enough to trigger it).

**Comment by shymega** [comment]
> @shymega tested the reproducibility of the multimonitor bug on the input-leap client. TLDR; mutter 45 rc-2 on the multi-monitors input leap client is the cause of the bug.
> 
> Downgrading mutter (libmutter-13-0 and its gir and common-bin companions) from 45.0 to 45-rc-2 on the Debian input-leap client reproduces the issue. All the other packages make no difference: gnome-shell, libei1, libeis1. Be it on the input-leap server or client. I can also have mutter 45-rc-2 on the input-leap server and mutter 45.0 on the input-leap client and then the bug is still not reproducible.
> 
> I also finally tried gnome-shell 45-rc1/mutter 45-rc-2 on both the client and server (as in my initial report) but this time with gnome-shell Xorg on the input-leap server and gnome-shell wayland on the input-leap server. The bug is still there (having mutter 45-rc-2 on the client is enough to trigger it).

Thanks - I've mentioned this in #109.

**Comment by whot** [comment]
@prahal your comment in https://github.com/input-leap/input-leap/issues/1673#issuecomment-1718243065 has a number of (valid) bugs but they're hidden here in a bug report with an ambiguous title. Can you please file separate issues for all those (and those in follow-up comments) so we can fix them one-by-one. Please cc me on the bugs as you file them, thanks.

**Comment by whot** [comment]
> Is the non-working clipboard sharing in wayland mode expected?

ftr, I haven't done any clipboard work for the libei mode (that one would actually require wayland), so that's expected to be broken. worth filing a bug either way to not forget about that

**Comment by prahal** [comment]
@whot about splitting the comment into dedicated bug reports, done a few days ago:
- about the input-leap client asking for the XDG portal right RemoteDesktop each time its screen blanks: https://github.com/input-leap/input-leap/issues/1699
- about missing support for Wayland clipboard sharing: https://github.com/input-leap/input-leap/issues/1698
- the bug with multi-monitor on input-leap client is fixed post mutter 45-rc-2 i.e. the final mutter 45.0 release is fine.

If I understood well, you mean there requires new libei and Wayland work for clipboard support ... I thought it was a matter of adding support to input-leap for wl-copy ...
then I should open a bug report for these projects too. libei for a start I guess?

**Comment by whot** [comment]
let's move the various topics to those bugs instead of here, makes them easier to track. I've already replied in those.

**Comment by shymega** [comment]
Concur. Closing this issue in favour of your suggestion.
